advanced_source/cpp_export.rst

```diff
@@ -0,0 +1,364 @@
+  -- Looking for pthread_create in pthreads
+  -- Looking for pthread_create in pthreads - not found
+  -- Looking for pthread_create in pthread
+  -- Looking for pthread_create in pthread - found
+  -- Found Threads: TRUE
+  -- Configuring done
+  -- Generating done
+  -- Build files have been written to: /example-app/build
+  root@4b5a67132e81:/example-app/build# make
+  Scanning dependencies of target example-app
+  [ 50%] Building CXX object CMakeFiles/example-app.dir/example-app.cpp.o
+  [100%] Linking CXX executable example-app
+  [100%] Built target example-app
+
+If we supply the path to the serialized ``ResNet18`` model we created earlier
+to the resulting ``example-app`` binary, we should be rewarded with a friendly
+"ok":
+
+.. code-block:: sh
+
+  root@4b5a67132e81:/example-app/build# ./example-app model.pt
+  ok
+
+Step 4: Executing the Script Module in C++
+----------------------------------------
+
+Having succesfully loaded our serialized ``ResNet18`` in C++, we are now just a
+couple lines of code away from executing it! Let's add those lines to our C++
+application's ``main()`` function:
+
+.. code-block:: cpp
+
+    // Create a vector of inputs.
+    std::vector<torch::jit::IValue> inputs;
+    inputs.push_back(torch::ones({1, 3, 224, 224}));
+
+    // Execute the model and turn its output into a tensor.
+    auto output = module->forward(inputs).toTensor();
+
+    std::cout << output.slice(/*dim=*/1, /*start=*/0, /*end=*/5) << '\n';
+
+The first two lines set up the inputs to our model. We create a vector of
+``torch::jit::IValue`` (a type-erased value type ``script::Module`` methods
+accept and return) and add a single input. To create the input tensor, we use
+``torch::ones()``, the equivalent to ``torch.ones`` in the C++ API.  We then
+run the ``script::Module``'s ``forward`` method, passing it the input vector we
+created. In return we get a new ``IValue``, which we convert to a tensor.
+
+.. tip::
+
+  To learn more about functions like ``torch::ones`` and the PyTorch C++ API in
+  general, refer to its documentation at https://pytorch.org/cppdocs.
+
+In the last line, we print the first five entries of the output. Since we
+supplied the same input to our model in Python earlier in this tutorial, we
+should ideally see the same output. Let's try it out by re-compiling our
+application and running it with the same serialized model:
+
+.. code-block:: sh
+
+  root@4b5a67132e81:/example-app/build# make
+  Scanning dependencies of target example-app
+  [ 50%] Building CXX object CMakeFiles/example-app.dir/example-app.cpp.o
+  [100%] Linking CXX executable example-app
+  [100%] Built target example-app
+  root@4b5a67132e81:/example-app/build# ./example-app model.pt
+  -0.2698 -0.0381  0.4023 -0.3010 -0.0448
+  [ Variable[CPUFloatType]{1,5} ]
+
+
+For reference, the output in Python previously was::
+
+  tensor([-0.2698, -0.0381,  0.4023, -0.3010, -0.0448], grad_fn=<SliceBackward>)
+
+Looks like a good match!
```

Comment:
Plug the C++ API here: if you need to do any preprocessing, basically all of PyTorch is available as a C++ API on these torch::Tensor objects.